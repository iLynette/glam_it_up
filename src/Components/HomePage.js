import { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchProductFromApi } from '../redux/Products/products';
import Categories from './Categories';
import Btn from './Arrow';

const HomePage = () => {
  const data = useSelector((state) => state.productReducer);
  const [value, setValue] = useState('');
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchProductFromApi());
  }, []);

  const handleClick = (e) => {
    setValue(e.target.value);
  };
  const filterProducts = data.filter((product) =>
    product.product_type.toLowerCase().includes(value.toLowerCase()));
  return (
    <>
    <Categories data={data} handleClick={handleClick} />
      <div className="products">
        {value === 'Select a product type'
         ? data.map((product) => (
          <div key={product.id}>
            <p className="head">
              {product.name}
              <Btn id={product.id} />
            </p>
            <p>{product.product_type}</p>
            <p>{product.price}</p>
            {/* <p key={product.id}>{product.description}</p> */}
            {/* <p key={product.id}>
              <img src={product.image_link} alt={product.title} />
            </p> */}
          </div>
        ))
        
      </div>
    </>
  );
};

export default HomePage;
