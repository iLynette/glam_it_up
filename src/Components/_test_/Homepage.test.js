import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import HomePage from '../HomePage';
import store from '../../redux/configStore';
const HomePageProvider = () => (
    <Provider store={store}>
        <HomePage />
    </Provider>
)

describe('The categories render properly', () => {
    it('renders correctly', () => {
        const component = renderer.create(<HomePageProvider />).toJSON()
        expect(component).toMatchSnapshot();
    })
})
