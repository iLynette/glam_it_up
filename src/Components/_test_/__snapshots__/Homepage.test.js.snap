// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The homepage render properly renders correctly 1`] = `
Array [
  <select
    className="items"
    onChange={[Function]}
    onClick={[Function]}
    required={true}
  >
    <option
      value="select a product type"
    >
      select a product type
    </option>
  </select>,
  <div
    className="products"
  />,
]
`;
